src/nostr_service.cpp

```diff
@@ -182,104 +182,108 @@ tuple<vector<string>, vector<string>> NostrService::publishEvent(shared_ptr<Even
     return make_tuple(successfulRelays, failedRelays);
 };
 
-// TODO: Make this method return a promise.
 // TODO: Add a timeout to this method to prevent hanging while waiting for the relay.
-vector<shared_ptr<Event>> NostrService::queryRelays(shared_ptr<Filters> filters)
+std::future<vector<shared_ptr<Event>>> NostrService::queryRelays(shared_ptr<Filters> filters)
```

Comment:
We should be able to just use `future` without the namespace in this file, since we have the using declarations at the top of the file.

```suggestion
future<vector<shared_ptr<Event>>> NostrService::queryRelays(shared_ptr<Filters> filters)
```